src/math/quaternion_operations.hpp

```diff
@@ -245,4 +305,30 @@ Kokkos::Array<double, 4> NormalizeQuaternion(const Kokkos::Array<double, 4>& q)
     return normalized_quaternion;
 }
 
+/**
+ * @brief Returns a 4-D quaternion from provided tangent vector and twist (degrees) about tangent
+ */
+inline Array_4 TangentTwistToQuaternion(const Array_3& tangent, const double twist) {
+    const auto e1 = UnitVector(tangent);
+    auto a = e1[0] > 0. ? 1. : -1.;
+    const Array_3 e2{
+        -a * e1[1] / sqrt(e1[0] * e1[0] + e1[1] * e1[1]),
+        a * e1[0] / sqrt(e1[0] * e1[0] + e1[1] * e1[1]),
```

Comment:
Couple comments
- This returns `NaN` when the tangent is the z-axis, i.e. `e1 = e2 = 0.` -- we should probably check for div by zero here
- We could simplify this by introducing a local variable as such
```
    const auto denominator = std::sqrt(e1[0] * e1[0] + e1[1] * e1[1]);
    const Array_3 e2{
        -a * e1[1] / denominator,
        a * e1[0] / denominator,
        0.,
    };
```